### ISSUE TITLE
Regular expression in `t.deepEqual()` shouldn't trigger warning

### ISSUE THREAD

**Issue by Airkro** [avajs/eslint-plugin-ava#270]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2440%20Funded-%2300A156.svg" />](https://issuehunt.io/r/avajs/eslint-plugin-ava/issues/270)

Rule: [no-incorrect-deep-equal](https://github.com/avajs/eslint-plugin-ava/blob/master/docs/rules/no-incorrect-deep-equal.md)

Example:
```js
const ext2regexp = require('ext-to-regexp');

const jsx = ext2regexp('js', 'jsx');

t.deepEqual(jsx, /\.(js|jsx)$/); //  shouldn't trigger warning
```

---

> *Note: This issue has a bounty, so it's expected that you are an experienced programmer and that you give it your best effort if you intend to tackle this. Don't forget, if applicable, to add tests, docs (double-check for typos). And don't be sloppy. Review your own diff multiple times and try to find ways to improve and simplify your code. Instead of asking too many questions, present solutions. The point of an issue bounty is to reduce my workload, not give me more. Include a 🦄 in your PR description to indicate that you've read this. Thanks for helping out 🙌* - @sindresorhus

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $40.00)

- [<img src='https://avatars3.githubusercontent.com/u/44827199?v=4' alt='issuehunt' width=24 height=24> issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)
### Submitted pull Requests
- [#276 do not trigger no-incorrect-deep-equal for regular expression](https://issuehunt.io/r/avajs/eslint-plugin-ava/pull/276)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/avajs/eslint-plugin-ava/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

**Comment by Airkro** [comment]
#158
